src/layout.js

```diff
@@ -434,6 +439,9 @@ export function applyStaticLayout(element) {
     if (height) {
       setStyle(element, 'height', height);
     }
+  } else if (layout == Layout.FLUID) {
+    element.classList.add('i-amphtml-layout-awaiting-size');
+    setStyle(element, 'width', width ? width : '100%');
```

Comment:
This will work well if there's no margin. Is this valid assumption for you here?

Reply:
I would assume no restrictions on margins. Would auto or inherit make more sense?

Reply:
Actually, after running some manual tests, I don't really see the issue with margins. If I put an ad slot inside of a div with margin-left:50%, safeframe is smart enough to resize the creative to fit the outer container without causing any overflow beyond the page. Was there another concern that you had?

Reply:
The problem could arise from `position:absolute` coming from `i-amphtml-layout-awaiting-size`. Here's an example: http://jsbin.com/pujeba/edit?html,css,output

Not sure yet how to make two boxes the same size yet, but I assume that'd be exactly what we'd want here, right?

Reply:
I style the iframe to use `position:relative` before attempting to resize, which I think takes care of this problem.

Reply:
That'd force the layout jump, right? We applied `position:absolute` before to ensure that layout stays untouched until the full resize is ready.

Reply:
Not sure what you mean by layout jump. Here's an image of an ad placed in a container with margin-left set to 50%. 
![fluid](https://user-images.githubusercontent.com/20518281/30493599-b3bc0428-9a12-11e7-993d-31078498240f.png)

Reply:
Let's look at the sequence. There's no way that I know of for an element to take the right available width, but not affect layout, such as sibling margins, etc. The best approximate is `position:absolute`. That, however, loses `width: auto`. `width: 100%` is not a horrible estimate, but what we really want is `width: calc(100% - marginLeft - marginRight)`, which I don't know how we could do easily.

If, as you describe, we first set `position:relative` before measuring, then the sequence is:
1. Static layout: apply `i-amphtml-layout-awaiting-size` with `position:absolute`. No relayout, so cost = 0. No layout effect. Nice. But width is not accurate as described above.
2. Switch to `position:relative`. This will trigger relayout, so cost = 1 full relayout. This is user-visible relayout - i.e. the actual sibling margins will change, things that were next to each other could jump under each other, etc. But width becomes accurate, so that'd allow to measure.
3. Resize to `height: calculated`. This will trigger relayout again, but probably only partial. This is obviously user-visible, though we take care of this already.

So, 2 & 3 can be combined (and reduce layout jumps and cost) if we calculate the accurate expected width based on the formula: `width: calc(100% - marginLeft - marginRight)`. What's more problematic than cost - (2) will already cause a user-visible UI jump even though the actual resize in (3) could be denied.

Reply:
Here's another dumb question: Can we do something like call getComputedStyle(element.parentElement) and use the width from the returned object? Is that function too computationally expensive for this purpose, or are there other problems with that approach?

Reply:
The width along is not enough. You also need margins that are kind of hard to calculate given margin suppression rules and such. There's also a question of siblings, e.g. when used with flexbox and other. But assuming we find a way, calling `getComputedStyle` is fine from inside the `layoutCallback`, but `vsync.measure` would need to be used.

So the question is: how can we calculate maximum width an element would be allowed to take in this place without overflowing the parent.